wremnants/theory_tools.py

```diff
@@ -320,7 +324,7 @@ def define_theory_corr(df, dataset_name, helpers, generators, modify_central_wei
             df = df.Define(f"{generator}Weight_tensor", helper, ["massVgen", "absYVgen", "ptVgen", "chargeVgen", "nominal_weight_uncorr"])
 
         if i == 0 and modify_central_weight:
-            df = df.Define("theory_corr_weight", f"{generator}Weight_tensor(0)/nominal_weight_uncorr")
+            df = df.Define("theory_corr_weight", f"nominal_weight_uncorr == 0 ? 0 : {generator}Weight_tensor(0)/nominal_weight_uncorr")
```

Comment:
Maybe should be a threshold rather than exactly zero

Reply:
The Weight_tensor was multiplied with "nominal_weight_uncorr" such that it should cancel out. But if it is exactly 0 the code crashes. Maybe a threshold in the order of the double precision would make sense

Reply:
I guess 0 is ok, for a threshold consistent with the double precision can use std::numeric_limits<double>::min() I think (not 100% sure if it's a float or double)